### DISCUSSION TITLE
Unrecognized syntax in an example

### DISCUSSION THREAD

**Discussion by karlmutch** [cue-lang/cue#1423] [Q&A] [answered]
When reading code in https://github.com/cue-lang/cue/issues/67 there is some syntax that uses gt, lt and somehow uses to populate a name field.

```
package multi

entry <Name>: {
	name: Name
}
entry Foo: { other: "blah" }
entry Bar: { field: "bleh" }
```
Followed in the discussion thread by 
```
command gen_out: {
    for x in entry {
	task "write_\(x.name)": file.Create & {
		filename: "\(_OUTDIR)/\(x.name).json"
		contents: json.Indent(json.Marshal(x), "", "  ")
	}
    }
}
```

I cannot make these snipets work at all.  Does anyone have pointers to documentation about what the \<Name\> is and how to iterate multiple entry items properly ?

Many Thanks

**Comment by slewiskelly** [comment, marked as answer]
I'm not sure, but maybe `<>` was the old syntax for [templates](https://cuelang.org/docs/tutorials/tour/types/templates/).

This is a working example using today's syntax:

```cue
package multi

import (
    "encoding/json"
    "tool/file"
)

_OUTDIR: "/tmp"

entry: [Name=string]: {} & {
	name: Name
}

entry: Foo: { other: "blah" }
entry: Bar: { field: "bleh" }

command: gen_out: {
    for x in entry {
	    task: "write_\(x.name)": file.Create & {
		    filename: "\(_OUTDIR)/\(x.name).json"
		    contents: json.Indent(json.Marshal(x), "", "  ")
	    }
    }
}
```
